### PR TITLE
fix(api): handle 4XX errors get-public-profile

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

There is probably a "Loopback" way to do this, but less LB3 == more better.

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
> Do we know how to quantify our own use of this endpoint?

If I understand:
Best: Build and serve client, then view a profile at `/<username>` (e.g. `/developmentuser`)
Easiest: `npm run develop`, go to `/developmentuser`, click `Go to profile`, view network requests as both signed-in user and incognito user

**Review by Nirajn2311 — COMMENTED** [review]
If I understood the PR correctly, then it remove the public profile endpoint from loopback User model and we create our own logic and handler right? If that's the case then this is good to go from mobile perspective as we're not using this endpoint.

**Review by raisedadead — APPROVED** [review]
LGTM

File: api-server/src/server/boot/randomAPIs.js [hunk 1]
```diff
@@ -105,4 +112,114 @@ module.exports = function (app) {
         .catch(next);
     });
   }
+
+  const blockedUserAgentParts = ['python', 'google-apps-script'];
```
Comment (raisedadead) [inline api-server/src/server/boot/randomAPIs.js:116]:
```suggestion
  const blockedUserAgentParts = ['python', 'google-apps-script', 'curl'];
```

File: api-server/src/server/boot/randomAPIs.js [hunk 2]
```diff
@@ -105,4 +112,114 @@ module.exports = function (app) {
...
+
+  function blockUserAgent(req, res, next) {
+    const userAgent = req.headers['user-agent'];
+
+    if (
+      !userAgent ||
+      blockedUserAgentParts.some(ua => userAgent.toLowerCase().includes(ua))
+    ) {
+      return res.status(403).send('Forbidden');
```
Comment (raisedadead) [inline api-server/src/server/boot/randomAPIs.js:125]:
```suggestion
      return res
        .status(400)
        .send(
          'This endpoint is no longer available outside of the freeCodeCamp ecosystem'
        );
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
Sorry - some last minute nitpicks

**Review by raisedadead — APPROVED** [review]

**Pull request merged**